wasi/wasi.go

```diff
@@ -170,6 +173,30 @@ func Args(args []string) (Option, error) {
 	}, nil
 }
 
+// Environ returns an option to set environmental variables to the WASIEnvironment.
+// * keys: the names of environment variables to set
+// * values: the corresponding values of environment variables to set
+// Environ returns an error if the length of the keys and values doesn't match,
+// or the length or the total size of those strings exceed the max of uint32.
+func Environ(keys []string, values []string) (Option, error) {
```

Comment:
`os.Environ` is []string, should we match that up for convenience (considering this is a go project). It also seems to match the representation. Moreover a map is a more idiomatic way for key/value if we were going for that. I think we should re-use how `os.Environ` does it.

Then, we can say `Note: while this signature matches os.Environ conventions, env propagation into WASI is intentionally done manually here`

wdyt?

Reply:
Yeah, I agree with the format of `os.Environ` is the best choice for ergonomics, considering that the common use cases are calling `os.Environ` to get the all environment variables, and pass it to this `wasi.Environ` (possibly with some filtering). I'd like to merge the design discussion on vs map interface here. wdyt @mathetake?

Reply:
BTW, why my first design was the same as that of wasmtime, but now I think the format of `os.Environ` is better than that for the reason above.